### ISSUE TITLE
Eliminate ActualZoneNum and move checks for controlled zone and plenum out of ZoneTempPredictorCorrector::CalcZoneSums

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6278]
Issue overview
--------------
- [x] In [ZoneTempPredictorCorrector::CalcZoneSums](https://github.com/NREL/EnergyPlus/blob/8856cdfd47cbc142b22bdea4dbc074029650e4de/src/EnergyPlus/ZoneTempPredictorCorrector.cc#L5072-L5097), every time step there are loops through all zones to set ControlledZoneAirFlag and ZoneEquipConfigNum and loops through all return and supply plenums to set ZoneRetPlenumAirFlag and ZoneSupPlenumAirFlag.  These should be done once somewhere else and stored in the Zone or ZoneEquipConfig data.  There's even an old comment in there suggesting this.  The performance improvement will likely be disappointing, but still should be done. 
*This part was addressed in #6272*

- [x] Also, it's time to get rid of the distinction between ControlledZoneNum and ActualZoneNum - all zone-based data structures should be parallel.  I'm fairly certain that ZoneEquipConfig and Zone and all similar arrays are all dimensioned to NumZones, so it's just a matter of making sure they all get filled in the same way.  ActualZoneNum and ZoneEqNum and CtrldZoneNum could all be collapsed into ZoneNum which would be less confusing.  
- [ ] Also look for `ControlledZoneIndex` which shows up in some functions.
- [ ] And this could eliminate the need for `GetControlledZoneIndex`

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus Version 8.8.0-8856cdfd47

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file n/a - change should be a no-diff change
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by lgu1234** [comment]
@mjwitte I looked at Item 1. Here are my opinions. 

CalcZoneSums is called 2 times. One is called by the predictor, and the other is called by the corrector. The following sums are calculated:
                     Real64 &SumIntGain, // Zone sum of convective internal gains
                      Real64 &SumHA,      // Zone sum of Hc*Area
                      Real64 &SumHATsurf, // Zone sum of Hc*Area*Tsurf
                      Real64 &SumHATref,  // Zone sum of Hc*Area*Tref, for ceiling diffuser convection correlation
                      Real64 &SumMCp,     // Zone sum of MassFlowRate*Cp
                      Real64 &SumMCpT,    // Zone sum of MassFlowRate*Cp*T
                      Real64 &SumSysMCp,  // Zone sum of air system MassFlowRate*Cp
                      Real64 &SumSysMCpT, // Zone sum of air system MassFlowRate*Cp*T

The last two items (SumSysMCp and SumSysMCpT) are calculated in the Corrector call only. I suggest to move the section to CorrectZoneAirTemp. The movement will be consistent with CorrectZoneHumRat.

I also observe that the first 6 items are constant from both Predictor and Corrector calls. I suggest to call once from the Predictor and save these results under Zone and use these results in the Corrector, so that no duplicated calls are needed.

It is possible to keep the following section, because AirflowNetworkExchangeData may be changed after the Predictor.

        // Sum all multizone air flow calculated from AirflowNetwork by assuming no simple air infiltration model
        if (SimulateAirflowNetwork == AirflowNetworkControlMultizone || SimulateAirflowNetwork == AirflowNetworkControlMultiADS ||
            (SimulateAirflowNetwork == AirflowNetworkControlSimpleADS && AirflowNetworkFanActivated)) {
            // Multizone airflow calculated in AirflowNetwork
            SumMCp = AirflowNetworkExchangeData(ZoneNum).SumMCp + AirflowNetworkExchangeData(ZoneNum).SumMMCp;
            SumMCpT = AirflowNetworkExchangeData(ZoneNum).SumMCpT + AirflowNetworkExchangeData(ZoneNum).SumMMCpT;
        }
 
In other words, this section may be called twice.

Please let me know what you think.

**Comment by mjwitte** [comment]
> I also observe that the first 6 items are constant from both Predictor and Corrector calls. I suggest to call once from the Predictor and save these results under Zone and use these results in the Corrector, so that no duplicated calls are needed.

@lgu1234 Is this true for surfaces with radiant heating systems? If I remember correctly, those system types call updates to the surface heat balance, so H and T could change between predictor and corrector.

**Comment by lgu1234** [comment]
@mjwitte I am not sure. I can assume radiant heating systems will be changed during HVAC simulations. It is better to call surface heat transfer twice.
